File: android_java_hummus/src/main/java/com/github/klee0kai/hummus/adapterdelegates/delegate/ViewAdapterDelegate.java
package com.github.klee0kai.hummus.adapterdelegates.delegate;

import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


import com.github.klee0kai.hummus.adapterdelegates.SimpleViewHolder;
import com.hannesdorfmann.adapterdelegates4.AdapterDelegate;

import java.util.List;

public class ViewAdapterDelegate extends AdapterDelegate<List<Object>> {


    @Override
    protected boolean isForViewType(@NonNull List<Object> items, int position) {
        return items.get(position) instanceof View;
    }

    @NonNull
    @Override
    protected RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent) {
        FrameLayout frameLayout = new FrameLayout(parent.getContext());
        frameLayout.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT));
        return new SimpleViewHolder(frameLayout);
    }

    @Override
    protected void onBindViewHolder(@NonNull List<Object> objects, int i, @NonNull RecyclerView.ViewHolder vh, @NonNull List<Object> list) {
        ViewGroup container = (ViewGroup) vh.itemView;
        View v = (View) objects.get(i);
        if (v.getParent() != null)
            ((ViewGroup) v.getParent()).removeView(v);

        container.removeAllViews();
        container.addView(v);
    }

}

File: android_kotlin_hummus/src/main/kotlin/com/github/klee0kai/hummus/adapterdelegates/delegate/ViewAdapterDelegate.kt
package com.github.klee0kai.hummus.adapterdelegates.delegate

import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.github.klee0kai.hummus.adapterdelegates.SimpleViewHolder
import com.hannesdorfmann.adapterdelegates4.AdapterDelegate

open class ViewAdapterDelegate(
    protected open val layoutParams: ViewGroup.LayoutParams = ViewGroup.LayoutParams(
        ViewGroup.LayoutParams.MATCH_PARENT,
        ViewGroup.LayoutParams.WRAP_CONTENT
    )
) : AdapterDelegate<List<Any>>() {

    constructor(@RecyclerView.Orientation orientation: Int) :
            this(
                when (orientation) {
                    RecyclerView.HORIZONTAL -> ViewGroup.LayoutParams(
                        ViewGroup.LayoutParams.WRAP_CONTENT,
                        ViewGroup.LayoutParams.MATCH_PARENT
                    )
                    else -> ViewGroup.LayoutParams(
                        ViewGroup.LayoutParams.MATCH_PARENT,
                        ViewGroup.LayoutParams.WRAP_CONTENT
                    )
                }
            )


    override fun isForViewType(items: List<Any>, position: Int): Boolean {
        return items[position] is View
    }

    override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
        val frameLayout = FrameLayout(parent.context)
        frameLayout.layoutParams = ViewGroup.LayoutParams(layoutParams)
        return SimpleViewHolder(frameLayout)
    }

    override fun onBindViewHolder(objects: List<Any>, i: Int, vh: ViewHolder, list: List<Any>) {
        val container = vh.itemView as ViewGroup

        // detach from current parent
        val view = objects[i] as View
        (view.parent as? ViewGroup)?.removeView(view)

        // attach to new parent
        container.apply {
            removeAllViews()
            addView(view)
        }
    }
}